rust/Cargo.toml

```diff
@@ -34,13 +34,16 @@ arrow-cast = "37.0.0"
 arrow-data = "37.0"
 arrow-ipc = { version = "37.0", features = ["zstd"] }
 arrow-ord = "37.0"
+arrow-row = "37.0"
 arrow-schema = "37.0"
 arrow-select = "37.0"
 async-recursion = "1.0"
 async-trait = "0.1.60"
 byteorder = "1.4.3"
 chrono = "0.4.23"
 clap = { version = "4.1.1", features = ["derive"], optional = true }
+# This is already used by datafusion
+dashmap = "5"
```

Comment:
Curious what is benefit of this lib over `std::HashMap` in this case

Reply:
dashmap allows multiple threads to insert concurrently, whereas std::HashMap doesn't allow concurrent inserts. This is just useful for building up the hashmap very fast, given we have the RHS data in memory already for most cases. And as mentioned in the comment, this is already used by DataFusion, so there's no incremental compilation cost from adding it to our dependencies.

Reply:
ok, i am fine with that. 

Minor question: we need to keep `dashmap` version sync with datafusion, right? Will cargo pull two different packages if we used the different versions?

Reply:
Yes, we do. They use 5 right now. We should check all of our dependencies for duplicates periodically, including tokio.